src/internal/converters/ics/ics.go

```diff
@@ -192,108 +199,148 @@ func getRecurrencePattern(
 }
 
 func FromJSON(ctx context.Context, body []byte) (string, error) {
-	data, err := api.BytesToEventable(body)
+	event, err := api.BytesToEventable(body)
 	if err != nil {
 		return "", clues.Wrap(err, "converting to eventable")
 	}
 
 	cal := ics.NewCalendar()
 	cal.SetProductId("-//Alcion//Corso") // Does this have to be customizable?
 
-	id := data.GetId() // XXX: iCalUId?
-	event := cal.AddEvent(ptr.Val(id))
+	id := ptr.Val(event.GetId())
+	iCalEvent := cal.AddEvent(id)
 
-	created := data.GetCreatedDateTime()
+	err = updateEventProperties(ctx, event, iCalEvent)
+	if err != nil {
+		return "", clues.Wrap(err, "updating event properties")
+	}
+
+	exceptionOcurrances := event.GetAdditionalData()["exceptionOccurrences"]
+	if exceptionOcurrances != nil {
+		for _, occ := range exceptionOcurrances.([]any) {
+			instance, ok := occ.(map[string]any)
+			if !ok {
+				return "", clues.New("converting exception instance to map[string]any").
+					With("type", fmt.Sprintf("%T", instance))
+			}
+
+			exBody, err := json.Marshal(instance)
+			if err != nil {
+				return "", clues.Wrap(err, "marshalling exception instance")
+			}
+
+			exception, err := api.BytesToEventable(exBody)
+			if err != nil {
+				return "", clues.Wrap(err, "converting to eventable")
+			}
+
+			exICalEvent := cal.AddEvent(id)
+			start := exception.GetOriginalStart() // will always be in UTC
+
+			exICalEvent.AddProperty(ics.ComponentProperty(ics.PropertyRecurrenceId), start.Format(iCalDateTimeFormat))
+
+			err = updateEventProperties(ctx, exception, exICalEvent)
+			if err != nil {
+				return "", clues.WrapWC(ctx, err, "updating exception event properties")
```

Comment:
WrapWC shouldn't be neededhere, since the func takes in a ctx.
```suggestion
				return "", clues.Wrap(err, "updating exception event properties")
```